src/gui/editorwidgets/core/qgseditorwidgetwrapper.cpp

```diff
@@ -49,6 +49,11 @@ QVariant QgsEditorWidgetWrapper::defaultValue() const
   return mDefaultValue;
 }
 
+void QgsEditorWidgetWrapper::resetValue()
+{
+  setValue( defaultValue() );
```

Comment:
Yes you're right, bad idea...

So, if a joined feature does not exist, I just update the corresponding widgets' values with NULL. Do you agree with that?